### ISSUE TITLE
[Msal 2.0.0-beta3] endless loop when the url is not window.location.origin.

### ISSUE THREAD

**Issue by qszhuan** [AzureAD/microsoft-authentication-library-for-js#1864]
## Library
- [x ] `@azure/msal-browser@2.0.0-beta3`

**Important**: Please fill in your exact version number above, e.g. `msal@1.1.3`.

## Description

If I open the application from `http://localhost:3000`, and enabled auto login, it works fine. but when I use `http://localhost:3000/anotherurl`, the login will be in an endless loop. 

I am not sure if I am doing it in a wrong way. basically I want the system login automatically, same as the `forceLogin` feature in `react-aad-msal` (https://www.npmjs.com/package/react-aad-msal).

## Security
- [ ] Is this issue security related?

## Regression
- [ x] Did this behavior work before? 
Version:
`Msal 1.3.2` with `react-aad-msal` works, the config I set for `Msal 1.3.2` is :

```js
import {MsalAuthProvider, LoginType} from 'react-aad-msal';

const config = {
  auth: {
    authority: process.env.REACT_APP_APP_AUTHORITY,
    clientId: process.env.REACT_APP_APP_ID,
    validateAuthority: true,
    postLogoutRedirectUri: '',
  },
  cache: {
    cacheLocation: 'sessionStorage',
    storeAuthStateInCookie: false,
  },
};

const authenticationParameters = {
  scopes: ['user.read'],
  redirectUri: window.location.origin,
};

const options = {
  loginType: LoginType.Redirect,
  tokenRefreshUri: window.location.origin,
};

const authProvider = new MsalAuthProvider(config, authenticationParameters, options);

export async function getAccessToken(requestedScopes = ['user.read']) {
  var accessTokenResponse = await authProvider
    .acquireTokenSilent({scopes: requestedScopes})
    .catch((err) => {
      console.error('Failed to get access token', err);
      return authProvider.acquireTokenRedirect({scopes: requestedScopes});
    });
  // const accessTokenResponse = await authProvider.getAccessToken();
  return accessTokenResponse.accessToken;
}

export {authProvider};
```

## MSAL Configuration
the code can be found in [issue 1846](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/1846)

the different is I set the `forceLogin` to `true` in `index.js` file:
```js
ReactDOM.render(
  <React.StrictMode>
     <App forceLogin={true} />
  </React.StrictMode>,
  document.getElementById('root')
);
```

## Reproduction steps
1. logout if loggedin.
2. type url `http://localhost:3000/anotherurl`, the path can be anything.
3. then there is an endless loop, I also noticed that the `tokenResponse` is none.

## Expected behavior
 should be able to login sucessfully, and redirect back the the initial url.
## Browsers/Environment
- [ x] Chrome

**Comment by tnorling** [comment]
@qszhuan This may be addressed by #1862 but to check that there's not something else going on here can you verify a couple things for me?
1. Is `http://localhost:3000/anotherurl` registered as a `redirectUri` in the app portal? Make sure anything you set as `redirectUri` is registered. If you want to redirect to a page that is not registered, set the `redirectUri` to something that is registered and initiate your `loginRedirect` from another page.
2. Are you awaiting the resolution of `handleRedirectPromise` before initiating your login? Another possible reason for what you're experiencing is that you may be initiating a new login before the previous one has completed.

**Comment by qszhuan** [comment]
Hi @tnorling, thanks for you reply, I checked my code,  I forgot to set the status and double check in the login useCallback. after I changed to this(see the comment lines), no error anymore:

```js
const [authenticationState, setAuthenticationState] = useState(AuthenticationState.UnAuthenticated);
const [error, setError] = useState(null);

const loginCallBack = useCallback(
  () => {
    //check the current process status, if already Authenticated or InProgress, do nothing.
    if(authenticationState !== AuthenticationState.UnAuthenticated || error){
      return;
    }
    setAuthenticationState(AuthenticationState.InProgress); //set the status to InProgress
    authProvider.login();
  },
  [authenticationState, error],
)

```
I only registered `http://localhost:3000` in the app portal. and my `navigateToLoginRequestUrl ` is `true`(default).
I log in with `http://localhost:3000/anotherurl`, but the url becomes `http://localhost:3000/#` after logged in. I am using `loginRedirect`, so will wait for your PR to be released :)

Thanks a lot.
